src/SignalR/server/Core/src/HubConnectionHandler.cs

```diff
@@ -59,10 +59,16 @@ HubDispatcher<THub> dispatcher
             _globalHubOptions = globalHubOptions.Value;
             _logger = loggerFactory.CreateLogger<HubConnectionHandler<THub>>();
             _userIdProvider = userIdProvider;
-            _dispatcher = dispatcher;
 
             _enableDetailedErrors = _hubOptions.EnableDetailedErrors ?? _globalHubOptions.EnableDetailedErrors ?? false;
             _maximumMessageSize = _hubOptions.MaximumReceiveMessageSize ?? _globalHubOptions.MaximumReceiveMessageSize;
+
+            _dispatcher = new DefaultHubDispatcher<THub>(
+                serviceScopeFactory,
+                new HubContext<THub>(lifetimeManager),
+                hubOptions,
+                globalHubOptions,
+                new Logger<DefaultHubDispatcher<THub>>(loggerFactory));
```

Comment:
Wow this is kinda gross 😄

Reply:
Should we also remove `HubDispatcher` from DI now that it's no longer used in DI?